File: cli/internal/cartographer/testdata/sample.js
// Sample JavaScript file for Cartographer AST extraction tests.

class DataProcessor {
  constructor(config) {
    this.config = config;
  }

  process(record) {
    return record;
  }
}

function loadConfig(path) {
  return { path };
}

const formatValue = (val) => String(val);

const helper = function namedHelper(x) {
  return x;
};

let counter = 0;

File: cli/internal/cartographer/testdata/sample.py
"""Sample Python file for Cartographer AST extraction tests."""


class DataProcessor:
    """Processes incoming data records."""

    def process(self, record):
        return record


class Config:
    """Configuration container."""

    def __init__(self, path):
        self.path = path


def load_config(path):
    """Load configuration from disk."""
    return Config(path)


def _internal_helper():
    """Private helper — should still be extracted (Python has no enforced access)."""
    pass
